app/models/proxy_order.rb

```diff
@@ -42,21 +42,8 @@ def resume
 
   def initialise_order!
     return order if order.present?
-    create_order!(
-      customer_id: subscription.customer_id,
-      email: subscription.customer.email,
-      order_cycle_id: order_cycle_id,
-      distributor_id: subscription.shop_id,
-      shipping_method_id: subscription.shipping_method_id
-    )
-    order.update_attribute(:user, subscription.customer.user)
-    subscription.subscription_line_items.each do |sli|
-      order.line_items.build(variant_id: sli.variant_id, quantity: sli.quantity, skip_stock_check: true)
-    end
-    order.update_attributes(bill_address: subscription.bill_address.dup, ship_address: subscription.ship_address.dup)
-    order.update_distribution_charge!
-    order.payments.create(payment_method_id: subscription.payment_method_id, amount: order.reload.total)
-
+    factory = OrderFactory.new(order_attrs, skip_stock_check: true)
+    self.order = factory.create
```

Comment:
Maybe a stupid question, but what is a ProxyOrder? Some class level documentation could help to understand this.

Reply:
I'll add a comment, to the top of the class, but for posterity I'll also describe it here.

A ProxyOrder is a placeholder for an order that allows us to lazily create real orders. Each subscription has many ProxyOrders, one for each OrderCycle to which the Subscription applies. Each ProxyOrder knows how to instantiate a real order when it needs to (ie, when a user wants to alter a specific items for a given order cycle, or when the order is actually placed when the order cycle opens. ProxyOrders can also be cancelled (without having to instantiate an actual Order and the cancel it).

Probably the biggest advantage of using the proxy pattern is that changes can be made to the underlying subscription (eg. payment or shipping methods changed, line items added) without having to propagate these changes to a bunch of instantiated orders (though of course this logic is required because orders CAN be instantiated, but use of proxy orders just reduces complexity as much as possible while maintaining flexibility).